### PR TITLE
libs: Add OPA Gatekeeper library

### PR DESCRIPTION
**Pull request by iainlane**
https://github.com/open-policy-agent/gatekeeper/

### CODE REVIEW

**Comment by iainlane** [comment]
Is there a good way to test the usability of the libraries or would you normally merge them and then iterate, add some custom code / fix up if they don't work so well in practice?

**Comment by jvrplmlmn** [comment]
> Is there a good way to test the usability of the libraries or would you normally merge them and then iterate, add some custom code / fix up if they don't work so well in practice?



I believe that following the steps here: https://github.com/jsonnet-libs/k8s#create-or-update-a-new-lib would leave you with a local copy of the generated library. If that is something that you want to experiment with, you could vendor it with your project.

**Review by julienduchesne — APPROVED** [review]
LGTM! Let's just merge it and see what happens

File: libs/gatekeeper/config.jsonnet [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  name='gatekeeper',
+  specs=[
+    {
+      local url = 'https://raw.githubusercontent.com/open-policy-agent/gatekeeper/v%s/charts/gatekeeper/crds' % v.version,
+      output: v.output,
+      prefix: '^sh\\.gatekeeper\\..*',
+      crds: [
+        '%s/external-secrets.io_clustersecretstores.yaml' % url,
+        '%s/external-secrets.io_externalsecrets.yaml' % url,
+        '%s/external-secrets.io_secretstores.yaml' % url,
```
Comment (jvrplmlmn) [inline libs/gatekeeper/config.jsonnet:17]: These CRDs seem to be related to external-secrets?

Comment (julienduchesne) [reply libs/gatekeeper/config.jsonnet:17]: Indeed, didn't read 😆 . And the lib is empty: https://github.com/jsonnet-libs/gatekeeper-libsonnet/tree/main/3.12

**Pull request merged**